CRM/Member/BAO/Membership.php

```diff
@@ -1549,38 +1549,17 @@ public static function deleteMembershipPayment($membershipId, $preserveContrib =
    *   Do we only want active ones?
    *   (probably this should default to TRUE but as a newly added parameter we are leaving default b
    *   behaviour unchanged).
+   * @param bool $recurOnly
+   *   Do we only want membership types that can auto-renew (recur)?
    *
    * @return array
+   * @throws \CiviCRM_API3_Exception
    */
-  public static function buildMembershipTypeValues(&$form, $membershipTypeID = array(), $activeOnly = FALSE) {
```

Comment:
Per https://github.com/civicrm/civicrm-core/pull/14919 I have reservations about adding another parameter here - I feel like we are adding complexity to a function that should be dying

Reply:
@eileenmcnaughton Wasn't completely keen myself but was trying to avoid another foreach iteration.  But I've dropped it so we can get this PR merged (and then the follow-up that you posted in https://github.com/civicrm/civicrm-core/pull/14919